include/deal.II/hp/dof_handler.h

```diff
@@ -968,12 +982,20 @@ namespace hp
     // these are the checks that correspond to the last block in the save()
     // function
     unsigned int n_cells;
-
-    ar &n_cells;
-
-    AssertThrow(n_cells == tria->n_cells(),
-                ExcMessage("The object being loaded into does not match the triangulation "
-                           "that has been stored previously."));
+    std::string  policy_name;
+
+    ar &n_cells &policy_name;
+
+    AssertThrow (n_cells == tria->n_cells(),
+                 ExcMessage ("The object being loaded into does not match the triangulation "
+                             "that has been stored previously."));
+    AssertThrow (policy_name == dealii::internal::policy_to_string(*policy),
+                 ExcMessage (std::string ("The policy currently associated with this DoFHandler (")
```

Comment:
Good point. I copied the code from `dofs/dof_handler.h`. I'll fix it in both places.